src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -446,6 +446,13 @@ public override IntermediateNode Visit(DottedNameNode node, IRTranslatorContext
                         }
                     }
                 }
+                else if (typeLhs.IsCustomObject)
+                {
+                    // Field access within a custom object.
+                    Contracts.Assert(typeLhs.IsCustomObject);
+
+                    result = new RecordFieldAccessNode(context.GetIRContext(node), left, nameRhs);
```

Comment:
This makes sense. If we go and add a GetField function later, we'll want to unify it at the IR level.